kasa/tests/test_smartdevice.py

```diff
@@ -310,15 +310,13 @@ async def test_modules_not_supported(dev: IotDevice):
 
 
 @device_smart
-async def test_update_sub_errors(dev: SmartDevice, caplog):
+async def test_try_get_response(dev: SmartDevice, caplog):
     mock_response: dict = {
-        "get_device_info": {},
-        "get_child_device_list": SmartErrorCode.PARAMS_ERROR,
+        "get_device_info": SmartErrorCode.PARAMS_ERROR,
     }
     caplog.set_level(logging.DEBUG)
-    with patch.object(dev.protocol, "query", return_value=mock_response):
-        await dev.update()
```

Comment:
As we don't try to access directly anything else besides the `get_device_info`, which has no default and raises an exception on errors, I adjusted this test to test the behavior of _try_get_response instead of that of the update().